docs/tech/https.md

```diff
@@ -0,0 +1,111 @@
+# HTTPS
+
+[HTTPS](https://en.wikipedia.org/wiki/HTTPS) secures information when it's
+being transmitted over the internet. Everybody should use HTTPS, but it is
+especially important that we use HTTPS because of the sensitive data we transmit.
+
+## Compatibility
+
+Some of this will be trial and error. Some settings can be tweaked on our side but
+some particularly old devices/software will just never work with our system.
+
+It's difficult to anticipate what devices are out there being used in large numbers,
+and what browsers are being used on those devices. In a number of cases devices with
+old operating systems (like Windows XP, old Android and old BlackBerry OS) will not
+be compatible with our HTTPS setup when using the default/built-in browser. If users
+have a recent version of a commonly used browser like Google Chrome, Mozilla Firefox,
+or Opera Browser installed then that should be compatible.
+
+Users accessing an HTTPS site via a recent version of Opera Mini should not have any
+compatibility issues on any platform but the effectiveness of the encrypted connection
+is reduced as the connection is decrypted at Opera's servers.
+
+### Server Name Indication (SNI) compatibility
+
+The first and most important feature that any device connecting to an HTTPS site on our
+cluster will need is Server Name Indication (SNI) support. This, unfortunately, cannot
+be worked around and is a strict requirement born out of the fundamental design of the
+cluster. Please see [the Wikipedia article](https://en.wikipedia.org/wiki/Server_Name_Indication#Suppor)
+for full details.
+
+The native/built-in browsers on the following mobile devices do not support SNI (although
+it may be possible for users to install a browser that does support SNI):
+
+- Android 2.3 "Gingerbread" and earlier
+- Nokia Symbian OS
+- BlackBerry OS 7.1 and earlier
+
+The following browsers on desktop devices do not support SNI:
+
+- Internet Explorer on Windows XP
+
+It's also important to note that the following Python software we use does not support SNI:
+
+- Any Python software running on Python < 2.7.9 (and note that the version of Python 2.7 provided
+  with Ubuntu 14.04 is 2.7.6. Also note that our Python Docker containers will always run the
+  latest versions of Python)
```

Comment:
Uurgh.. this is kind of complicated and my understanding of this has evolved a bit. It's not _any_ software using Python < 2.7.9, it's specifically the `ssl` module in the Python stdlib < 2.7.9. Which means a lot of things, e.g. `requests` won't work. But some things, e.g. modern `Twisted` do work 🙃 

I don't know how much of a problem this will be in the future as we move to newer Pythons and people stop running Ubuntu 14.04. But it has bitten us in the past.

Reply:
Haha I don't know much about this but I'll take your word for it